tests/core/test_task.py

```diff
@@ -825,3 +825,16 @@ def run(self, x):
         "If you're trying to run this task outside of a Flow context, "
         f"you need to call {run_call}" in str(exc_info)
     )
+
+
+def test_task_call_with_self_succeeds():
+    from pathlib import Path
+
+    seconds_task = task(
+        Path.is_absolute, target="{{task_slug}}_{{map_index}}", result=LocalResult()
+    )
+    initial = Path("foo")
+
+    with Flow("test") as flow:
+        seconds_task(initial)
+        assert flow.run().is_successful()
```

Comment:
It looks like you're using `Path.is_absolute` as a method here to show how `self` needs special handling, but I'm worried that won't be apparent to someone reading this in the future. Can you define a simple class with a single method instead?